dimod/higherorder.py

```diff
@@ -23,8 +23,92 @@
 from dimod.binary_quadratic_model import BinaryQuadraticModel
 from dimod.decorators import vartype_argument
 from dimod.vartypes import Vartype
+import dimod
+from dimod.response import SampleSet
+import numpy as np
 
-__all__ = ['make_quadratic']
+__all__ = ['make_quadratic','HigherOrderComposite']
+
+
+class HigherOrderComposite(dimod.ComposedSampler):
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param['penalty_strength'] = []
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, h, J, offset=0, penalty_strength=1.0, **parameters):
```

Comment:
this should be `sample_ising` to use the`ComposedSampler` ABC.

I do like that it's a composite. If it is used nested in a stack than in some sense it just becomes identity.

Reply:
so I kinda realized that we end up cheating here...
the main function name is now sample_ising but under the hood we are calling sample for the children.

do all composites have self.sample for sure?